src/main/java/com/eternalcode/combat/fight/controller/FightUnTagController.java

```diff
@@ -25,13 +28,13 @@ public FightUnTagController(FightManager fightManager, PluginConfig config, Noti
     @EventHandler
     void onPlayerDeath(PlayerDeathEvent event) {
         Player player = event.getEntity();
+        Player killer = player.getKiller();
 
-        if (!this.fightManager.isInCombat(player.getUniqueId())) {
-            return;
-        }
+        this.untagPlayer(player);
 
-        this.announcer.sendMessage(player, this.config.messages.playerUntagged);
-        this.fightManager.untag(player.getUniqueId());
+        if (this.config.settings.shouldReleaseAttacker) {
```

Comment:
W sumie to teraz możesz tutaj dać checka killer != null i untag metodę zrobić bez nullable player